### PR TITLE
Improved: Batch Modal component UI to configure brokering job before scheduling

### PR DESCRIPTION
**Pull request by shashwatbangar**
…scheduling(job-manager/148)

### Related Issues

https://github.com/hotwax/job-manager/issues/148

### In Progress
727d2b0d70e75faaccb61818150022aa296e583f
### Short Description and Why It's Useful

Now user will be to configure brokering job before scheduling

#### Before
Screenshot 2022-05-17 at 2 52 31 PM

#### After
Screenshot 2022-05-17 at 7 02 55 PM

### Contribution and Currently Important Rules Acceptance

- [Yes] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: src/components/BatchModal.vue [hunk 1]
```diff
@@ -6,26 +6,43 @@
...
+      <ion-input :placeholder="showCurrentDateTime()" v-model="jobName" />
     </ion-item>
+
+    <ion-item>
+      <ion-label>{{ $t('Order queue') }}</ion-label>
+        <ion-select slot="end" interface="popover" value="brokering-queue">
+          <ion-select-option value="brokering-queue">{{ $t("Brokering queue") }}</ion-select-option>
+          <ion-select-option value="preorder-parking">{{ $t("Pre-order parking") }}</ion-select-option>
+          <ion-select-option value="backorder parking">{{ $t("Back-order parking") }}</ion-select-option>
+        </ion-select>
```
Comment (azkyakhan) [inline src/components/BatchModal.vue:26]: Improve indentation

File: src/components/BatchModal.vue [hunk 2]
```diff
@@ -6,26 +6,43 @@
...
+
+    <ion-item>
+      <ion-label position="fixed">{{ $t('Unfillable') }}</ion-label>
+      <ion-toggle @ionChange="toggleChange" slot="end" color="secondary" />
+    </ion-item>
+
     <ion-item>
       <ion-label position="fixed">{{ $t("Schedule") }}</ion-label>
-      <ion-datetime :value="currentBatch?.runTime ? getDateTime(currentBatch.runTime) : ''" @ionChange="updateRunTime($event, currentBatch)" display-format="MM/DD/YYYY" placeholder="date picker" />
+      <ion-datetime class="ionic-datetime-component" color="secondary" presentation="time" />
```
Comment (azkyakhan) [inline src/components/BatchModal.vue:36]: Improve class name

File: src/components/BatchModal.vue [hunk 3]
```diff
@@ -133,4 +164,10 @@ export default defineComponent({
     };
   },
 });
-</script>
\ No newline at end of file
+</script>
+
+<style scope>
+.ionic-datetime-component {
```
Comment (azkyakhan) [inline src/components/BatchModal.vue:170]: What was the need to set the width of the date-time component?

**Review by azkyakhan — CHANGES_REQUESTED** [review]

File: src/components/BatchModal.vue [hunk 4]
```diff
@@ -6,26 +6,43 @@
           <ion-icon slot="icon-only" :icon="closeOutline" />
         </ion-button>
       </ion-buttons>
-      <ion-title>{{ currentBatch?.jobName ? currentBatch?.jobName : $t('New Batch') }}</ion-title>
+      <ion-title>{{ currentBatch?.jobName ? currentBatch?.jobName : $t('New broker run') }}</ion-title>
     </ion-toolbar>
   </ion-header>
 
   <ion-content>
+
```
Comment (azkyakhan) [inline src/components/BatchModal.vue:14]: Remove unwanted spacing

File: src/components/BatchModal.vue [hunk 5]
```diff
@@ -6,26 +6,43 @@
...
     </ion-item>    
 
-    <ion-fab vertical="bottom" horizontal="end" slot="fixed" @click="updateJob()">
+    <ion-fab @click="sendModalData(toggleValue, orderQueueValue)" vertical="bottom" horizontal="end" slot="fixed">
       <ion-fab-button>
         <ion-icon :icon="checkmarkDoneOutline" />  
       </ion-fab-button>
     </ion-fab>
   </ion-content>
+
```
Comment (azkyakhan) [inline src/components/BatchModal.vue:45]: Remove unwanted spacing

File: src/components/BatchModal.vue [hunk 6]
```diff
@@ -41,7 +58,10 @@ import {
   IonItem,
```
Comment (azkyakhan) [inline src/components/BatchModal.vue:58]: Arrange all the imported and exported UI components alphabetically.

File: src/components/BatchModal.vue [hunk 7]
```diff
@@ -121,15 +146,25 @@ export default defineComponent({
...
+    },
+    // sendModalData(toggleValue: boolean, orderQueueValue: ) {
+
+    // }
   },
   setup() {
     const store = useStore();
 
     return {
       closeOutline,
```
Comment (azkyakhan) [inline src/components/BatchModal.vue:165]: Arrange returned ionicons alphabetically.

**Review by azkyakhan — CHANGES_REQUESTED** [review]

File: .env.example [hunk 8]
```diff
@@ -9,3 +9,4 @@ VUE_APP_PRD_JOB_ENUMS={"IMP_PRDTS":"JOB_IMP_PROD_NEW","SYNC_PRDTS":"JOB_IMP_PROD
 VUE_APP_INV_JOB_ENUMS={"HARD_SYNC":"JOB_UL_INV","BOPIS_CORRECTION":"JOB_BOPIS_RALC"}
 VUE_APP_INITIAL_JOB_ENUMS={"IMP_PRDTS_BLK":"JOB_IMP_PROD_NEW_BLK","IMP_ORDERS_BLK":"JOB_IMP_ORD_BLK"}
 VUE_APP_BASE_URL=
+VUE_APP_BATCH_JOB_ENUMS={"JOB_BKR_ORD_UNF":{"id": "JOB_BKR_ORD_UNF","facilityId":"_NA_","unfillable":"true"},"JOB_BKR_ORD":{"id": "JOB_BKR_ORD","facilityId":"_NA_","unfillable":"false"},"JOB_BKR_PREORD_UNF":{"id":"JOB_BKR_PREORD_UNF","facilityId":"_NA_","unfillable":"true"},"JOB_BKR_PREORD":{"id":"JOB_BKR_PREORD","facilityId":"_NA_","unfillable":"false"},"JOB_BKR_BACKORD_UNF":{"id":"JOB_BKR_BACKORD_UNF","facilityId":"_NA_","unfillable":"true"},"JOB_BKR_BACKORD":{"id":"JOB_BKR_BACKORD","facilityId":"_NA_","unfillable":"false"},}
```
Comment (ymaheshwari1) [inline .env.example:13]: Update facility id for preorder and backorder data

File: src/locales/en.json [hunk 9]
```diff
@@ -74,6 +76,7 @@
   "Logout": "Logout",
   "More options": "More options",
   "New Batch": "New Batch",
```
Comment (ymaheshwari1) [inline src/locales/en.json:78]: Remove `New Batch` from en.json if not used.

File: src/components/BatchModal.vue [hunk 10]
```diff
@@ -6,21 +6,39 @@
           <ion-icon slot="icon-only" :icon="closeOutline" />
         </ion-button>
       </ion-buttons>
-      <ion-title>{{ currentBatch?.jobName ? currentBatch?.jobName : $t('New Batch') }}</ion-title>
+      <ion-title>{{ currentBatch?.jobName ? currentBatch?.jobName : $t('New broker run') }}</ion-title>
     </ion-toolbar>
   </ion-header>
 
-  <ion-content>
+  <ion-content>
```
Comment (ymaheshwari1) [inline src/components/BatchModal.vue:13]: Remove unwanted change

File: src/components/BatchModal.vue [hunk 11]
```diff
@@ -121,14 +161,18 @@ export default defineComponent({
       if(batch) {
         batch['runTime'] = DateTime.fromISO(ev['detail'].value).toMillis()
       }
-    }
+    },
+    showCurrentDateTime() {
+      const today = new Date();
```
Comment (ymaheshwari1) [inline src/components/BatchModal.vue:166]: We won't use the browser Date class instead use luxon DateTime to get the date

File: src/store/modules/job/actions.ts [hunk 12]
```diff
@@ -286,31 +286,40 @@ const actions: ActionTree<JobState, RootState> = {
...
-        batchBrokeringJobs.push({
-          ...job,
-          id: job.jobId,
-          frequency: job.tempExprId,
-          enumId: job.systemJobEnumId,
-          status: job.statusId
-        })
+      const batchJobEnums = JSON.parse(process.env?.VUE_APP_BATCH_JOB_ENUMS as string)
+      let batchJobEnumIds = Object.values(batchJobEnums)?.map((job: any) => { 
+        return job.id;
```
Comment (ymaheshwari1) [inline src/store/modules/job/actions.ts:291]: We can remove the return statement and instead just directly use job.id

File: src/store/modules/job/getters.ts [hunk 13]
```diff
@@ -9,6 +9,20 @@ const getters: GetterTree <JobState, RootState> = {
     getPendingJobs (state){
       return state.pending.list;
     },
+    getOrderBatchJobs (state){
+      const batchJobEnums = JSON.parse(process.env?.VUE_APP_BATCH_JOB_ENUMS as string)
+      const batchJobEnumIds = Object.values(batchJobEnums)?.map((job: any) => { 
+        return job.id;
```
Comment (ymaheshwari1) [inline src/store/modules/job/getters.ts:15]: Same no need to use return here.

File: src/store/modules/job/getters.ts [hunk 14]
```diff
@@ -9,6 +9,20 @@ const getters: GetterTree <JobState, RootState> = {
...
+      });
+
+      return batchJobEnumIds.reduce((batches: any, batchJobEnumId: string) => {
+          const jobs = state.cached[batchJobEnumId];
```
Comment (ymaheshwari1) [inline src/store/modules/job/getters.ts:19]: Improve indentation.

File: src/views/Orders.vue [hunk 15]
```diff
@@ -329,7 +326,14 @@ export default defineComponent({
...
     this.store.dispatch("job/fetchJobs", {
       "inputFields":{
-        "systemJobEnumId": Object.values(this.jobEnums),
+        "systemJobEnumId": Object.values(this.jobEnums).map((jobEnum: any) => jobEnum.id),
+        "systemJobEnumId_op": "in"
+      }
+    });
+    this.store.dispatch("job/fetchJobs", {
+      "inputFields":{
+        // "systemJobEnumId": ['JOB_BKR_ORD'],
```
Comment (ymaheshwari1) [inline src/views/Orders.vue:335]: Remove this comment.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/BatchModal.vue [hunk 16]
```diff
@@ -50,28 +72,37 @@ import { closeOutline, checkmarkDoneOutline } from 'ionicons/icons';
 import { mapGetters, useStore } from 'vuex';
 import { DateTime } from 'luxon';
 import { isFutureDate } from '@/utils';
+import { JobService } from '@/services/JobService';
```
Comment (ymaheshwari1) [inline src/components/BatchModal.vue:75]: Remove unused import.

File: src/components/BatchModal.vue [hunk 17]
```diff
@@ -96,14 +127,23 @@ export default defineComponent({
       modalController.dismiss({ dismissed: true });
     },
     async updateJob() {
-      const job = this.currentBatch ? this.currentBatch : this.getJob(this.enum)?.find((job: any) => job.status === 'SERVICE_DRAFT');
+      let batchJobEnum =  this.enum;
```
Comment (ymaheshwari1) [inline src/components/BatchModal.vue:130]:
```suggestion
      let batchJobEnum = this.enum;
```

File: src/components/BatchModal.vue [hunk 18]
```diff
@@ -96,14 +127,23 @@ export default defineComponent({
...
+
+      if (!batchJobEnum) {
+        const jobEnum: any = Object.values(this.jobEnums)?.find((job: any) => { 
+          return job.unfillable == this.unfillableOrder && job.facilityId == this.batchOrderId
```
Comment (ymaheshwari1) [inline src/components/BatchModal.vue:134]: Use strict equality `===`

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/BatchModal.vue (hunk 18 above)
Comment (shashwatbangar) [reply src/components/BatchModal.vue:134]: Sure sir!

File: src/locales/en.json (hunk 9 above)
Comment (shashwatbangar) [reply src/locales/en.json:78]: Done sir!

File: src/components/BatchModal.vue [hunk 19]
```diff
@@ -6,21 +6,39 @@
...
-  <ion-content>
+  <ion-content> 
     <ion-item>
       <ion-label position="fixed">{{ $t('Batch name') }}</ion-label>
-      <ion-input :placeholder="$t('New Batch')" v-model="jobName" />
+      <ion-input :placeholder="currentDateTime = showCurrentDateTime()" v-model="jobName" />
     </ion-item>
+    <ion-item>
+      <ion-label>{{ $t('Order queue') }}</ion-label>
+      <ion-select slot="end" interface="popover" v-model="batchOrderId">
```
Comment (adityasharma7) [inline src/components/BatchModal.vue:20]:
```suggestion
      <ion-select slot="end" interface="popover" v-model="batchFacilityId">
```

File: src/components/BatchModal.vue [hunk 20]
```diff
@@ -121,14 +160,17 @@ export default defineComponent({
       if(batch) {
         batch['runTime'] = DateTime.fromISO(ev['detail'].value).toMillis()
       }
-    }
+    },
+    getCurrentDateTime() {
+      return DateTime.now().toLocaleString(DateTime.DATETIME_MED);
```
Comment (adityasharma7) [inline src/components/BatchModal.vue:165]: set timezone from user profile

File: src/store/modules/job/getters.ts [hunk 21]
```diff
@@ -3,42 +3,53 @@ import JobState from './JobState'
 import RootState from '../../RootState'
 
 const getters: GetterTree <JobState, RootState> = {
-    getJobStatus: (state) => (id: string): any  => {
```
Comment (adityasharma7) [inline src/store/modules/job/getters.ts:6]: Revert indentation changes

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/job/getters.ts (hunk 21 above)
Comment (shashwatbangar) [reply src/store/modules/job/getters.ts]: Done sir!

File: src/components/BatchModal.vue (hunk 20 above)
Comment (shashwatbangar) [reply src/components/BatchModal.vue:165]: Sure sir! Done.

File: src/components/BatchModal.vue (hunk 19 above)
Comment (shashwatbangar) [reply src/components/BatchModal.vue]: Fixed variable name.

File: src/components/BatchModal.vue (hunk 16 above)
Comment (shashwatbangar) [reply src/components/BatchModal.vue]: Done sir!

File: src/views/Orders.vue (hunk 15 above)
Comment (shashwatbangar) [reply src/views/Orders.vue]: Removed.

File: src/store/modules/job/getters.ts (hunk 14 above)
Comment (shashwatbangar) [reply src/store/modules/job/getters.ts]: Done sir.

File: src/store/modules/job/actions.ts (hunk 12 above)
Comment (shashwatbangar) [reply src/store/modules/job/actions.ts]: Fixed it as your directed sir.

File: src/store/modules/job/getters.ts (hunk 13 above)
Comment (shashwatbangar) [reply src/store/modules/job/getters.ts]: Done sir

File: src/components/BatchModal.vue (hunk 10 above)
Comment (shashwatbangar) [reply src/components/BatchModal.vue]: Done sir!

File: .env.example (hunk 8 above)
Comment (shashwatbangar) [reply .env.example]: Yes sir done!

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**